cpp/src/join/joining.cu

```diff
@@ -153,87 +126,82 @@ gdf_error trivial_full_join(
     gdf_column_view( left_result, l_ptr, nullptr, result_size, dtype);
     gdf_column_view(right_result, r_ptr, nullptr, result_size, dtype);
     CUDA_CHECK_LAST();
-    return GDF_SUCCESS;
 }
 
 /* --------------------------------------------------------------------------*/
 /** 
  * @brief  Computes the join operation between two sets of columns
  * 
  * @param[in] num_cols The number of columns to join
- * @param[in] leftcol The left set of columns to join
- * @param[in] rightcol The right set of columns to join
+ * @param[in] leftcols  cudf table of left columns to join
+ * @param[in] rightcols cudf table of right  columns to join
  * @param[out] left_result The join computed indices of the left table
  * @param[out] right_result The join computed indices of the right table
  * @param[in] join_context A structure that determines various run parameters, such as
  *                         whether to perform a hash or sort based join
  * @tparam join_type The type of join to be performed
  * 
- * @returns GDF_SUCCESS upon succesfull compute, otherwise returns appropriate error code
+ * @returns void
  */
 /* ----------------------------------------------------------------------------*/
 template <JoinType join_type>
-gdf_error join_call( int num_cols, gdf_column **leftcol, gdf_column **rightcol,
+void join_call( int num_cols, cudf::table const& leftcols, cudf::table const& rightcols,
                      gdf_column *left_result, gdf_column *right_result,
                      gdf_context *join_context)
 {
-  GDF_REQUIRE( 0 != num_cols, GDF_DATASET_EMPTY);
-  GDF_REQUIRE( nullptr != leftcol, GDF_DATASET_EMPTY);
-  GDF_REQUIRE( nullptr != rightcol, GDF_DATASET_EMPTY);
-  GDF_REQUIRE( nullptr != join_context, GDF_INVALID_API_CALL);
+  CUDF_EXPECTS( 0 != num_cols, "Dataset is empty");
+  CUDF_EXPECTS( 0 != leftcols.num_columns(), "Left Dataset is empty");
```

Comment:
This will just compute indices for the joined columns and nothing else.

So, if the there were empty input, it would be returned earlier in the function call. 

This check is just to ensure that there will not be any coding issues in the main function.